File: core/identity/session_identity_engine.py
from __future__ import annotations

from typing import Any, Dict


def attach_identity_to_session(
    session: Dict[str, Any],
    identity: Dict[str, Any],
) -> Dict[str, Any]:
    merged = dict(session)
    merged["browser_identity"] = dict(identity)
    merged["identity_attached"] = True
    merged["bounded"] = True
    return merged


def restore_identity_session(
    session: Dict[str, Any],
) -> Dict[str, Any]:
    identity = dict(session.get("browser_identity", {}))

    return {
        "session": session,
        "identity": identity,
        "restored": bool(identity),
        "bounded": True,
    }

File: src/identity/sessionIdentityEngine.ts
/**
 * Converted from Python: core/identity/session_identity_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function attachIdentityToSession(session: any, identity: any): any {
  var merged: any = py.pyDict(session);
  py.setItem(merged, "browser_identity", py.pyDict(identity));
  py.setItem(merged, "identity_attached", true);
  py.setItem(merged, "bounded", true);
  return merged;
}
export function restoreIdentitySession(session: any): any {
  var identity: any = py.pyDict(py.get(session, "browser_identity", {}));
  return {"session": session, "identity": identity, "restored": py.truthy(identity), "bounded": true};
}
